src/components/delete-blog-post-widget/DeleteBlogPostWidget.tsx

```diff
@@ -10,28 +10,17 @@ interface DeleteBlogPostWidgetProps {
 }
 
 export const DeleteBlogPostWidget: React.FC<DeleteBlogPostWidgetProps> = ({ id }) => {
-  const utils = api.useContext();
-
-  const deleteBlogPostMutation = api.blogPost.delete.useMutation({
-    onSuccess(data, variables, context) {
-      return utils.blogPost.get.invalidate();
-    },
-  });
-
+  const deleteBlogPostMutation = api.blogPost.delete.useMutation();
   const [isModalOpen, setModalOpen] = useState(false);
   const [isSuccessful, setSuccessful] = useState(false);
 
   useEffect(() => {
     if (deleteBlogPostMutation.isSuccess) {
       setSuccessful(true);
-
       const timeoutId = setTimeout(() => {
-        setModalOpen(false);
-        setSuccessful(false);
+        location.reload();
```

Comment:
My first approach was to close only the Delete modal. However as Harold pointed out above, the Comment modal was still not closing. To close the Comment modal, I tried passing state from the parent `BlogPostViewer.tsx` down to the delete button, where after the Delete click I would set the state so that the parent would know to close the `CommentModal.tsx` component. But this approach did not end up working and I'm unsure why, so in the PR description I suggested reload in the interest of time.

Reply:
Instead of passing the state down, you can pass down a handler, something like onAfterDelete, that will be called after the delete operation is successful, and is implemented to close the parent modal. I am ok with the current approach also, so feel free to merge